### PR TITLE
Upgrade phpstan and deprecated code checking dependencies to 1.x

### LINKED ISSUE #69: Upgrade phpstan and related dependencies for PHP 8.1 support
From https://github.com/localgovdrupal/localgov_project/issues/65#issuecomment-1095145300

To use PHP 8.1 we need to upgrade phpstan and some related packages to version 1.x. The complete list is:

   - phpstan/phpstan
   - phpstan/phpstan-deprecation-rules
   - mglaman/phpstan-drupal

Before these updates can be merged the following issues need fixing:

- https://github.com/localgovdrupal/localgov_alert_banner/issues/198
- https://github.com/localgovdrupal/localgov_core/issues/126
- https://github.com/localgovdrupal/localgov_demo/issues/75
- https://github.com/localgovdrupal/localgov_directories/issues/181
- https://github.com/localgovdrupal/localgov_geo/issues/51
- https://github.com/localgovdrupal/localgov_guides/issues/82
- https://github.com/localgovdrupal/localgov_news/issues/54
- https://github.com/localgovdrupal/localgov_openreferral/issues/30
- https://github.com/localgovdrupal/localgov_services/issues/153
- https://github.com/localgovdrupal/localgov_subsites/issues/79
- https://github.com/localgovdrupal/localgov_workflows/issues/33
- https://github.com/localgovdrupal/localgov_theme/issues/206
-

### PR DESCRIPTION
**Pull request by stephen-cox**
Closes #69

### CODE REVIEW

**Comment by finnlewis** [comment]
So once this is merged: https://github.com/localgovdrupal/localgov_core/pull/128 the tests should pass.

**Comment by stephen-cox** [comment]
This can be merged once the following PRs are merged:

- https://github.com/localgovdrupal/localgov_core/pull/128
- https://github.com/localgovdrupal/localgov_demo/pull/76
- https://github.com/localgovdrupal/localgov_directories/pull/182
- https://github.com/localgovdrupal/localgov_geo/pull/52
- https://github.com/localgovdrupal/localgov_guides/pull/83
- https://github.com/localgovdrupal/localgov_news/pull/56
- https://github.com/localgovdrupal/localgov_openreferral/pull/31
- https://github.com/localgovdrupal/localgov_services/pull/155
- https://github.com/localgovdrupal/localgov_workflows/pull/34

**Review by finnlewis — APPROVED** [review]
Tests pass, please release me!!

**Pull request merged**
